test/common/upstream/upstream_impl_test.cc

```diff
@@ -211,6 +211,49 @@ TEST_P(StrictDnsParamTest, ImmediateResolve) {
   EXPECT_EQ(2UL, cluster.prioritySet().hostSetsPerPriority()[0]->healthyHosts().size());
 }
 
+TEST_P(StrictDnsParamTest, DropOverLoadConfigTest) {
```

Comment:
Can you please add tests that validate config correctness?
Also consider adding tests that control the randomness (i.e., similar to `EXPECT_CALL(random_, random()).WillRepeatedly(Return(1));`).

Reply:
Yes, added the tests to cover incorrect configurations as well. 
However, I do not understand why need to mock random(). The random() function is already tested by other code.

Overall, if we mock random as: EXPECT_CALL(random_, random()).WillRepeatedly(Return(1));, it will be essentially same as set the drop_overload value to 0.